sdk/storage/ustorageclient/samples/blob_append_bytes/main.c

```diff
@@ -70,7 +70,7 @@ void main(void)
         (void)printf("Uploaded Data:\n%s\n", data);
         (void)printf("Downloaded Data:\n%.*s\n", (int)file_size, file);
         // compare
-        if(strncmp(data, file, sizeof(data)) == 0)
+        if(strncmp((const char *)data, (const char *)file, sizeof(data)) == 0)
```

Comment:
why is this needed?

Reply:
compilation fails on Mac if casting is not added:
```
[ 97%] Building C object sdk/storage/ustorageclient/samples/blob_append_bytes/CMakeFiles/blob_append_bytes_exe.dir/main.c.o
/Users/vsts/agent/2.155.1/work/1/s/sdk/storage/ustorageclient/samples/blob_append_bytes/main.c:73:20: error: passing 'const unsigned char [12]' to parameter of type 'const char *' converts between pointers to integer types with different sign [-Werror,-Wpointer-sign]
        if(strncmp(data, file, sizeof(data)) == 0)
                   ^~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/string.h:84:26: note: passing argument to parameter '__s1' here
int      strncmp(const char *__s1, const char *__s2, size_t __n);
                             ^
/Users/vsts/agent/2.155.1/work/1/s/sdk/storage/ustorageclient/samples/blob_append_bytes/main.c:73:26: error: passing 'unsigned char [1024]' to parameter of type 'const char *' converts between pointers to integer types with different sign [-Werror,-Wpointer-sign]
        if(strncmp(data, file, sizeof(data)) == 0)
                         ^~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/string.h:84:44: note: passing argument to parameter '__s2' here
int      strncmp(const char *__s1, const char *__s2, size_t __n);
                                               ^
2 errors generated.

```

I added it to make Mac pipeline works. And this is the reason I would like to have the same Warnings on Linux/Windows